components/webgpu/wgpu_thread.rs

```diff
@@ -851,10 +900,14 @@ impl WGPU {
                         device_id,
                         command_encoder_id,
                     } => {
+                        let pass = self
+                            .compute_passes
+                            .get_mut(&compute_pass_id)
+                            .expect("ComputePass should exists");
                         // TODO: Command encoder state error
-                        if let Some((mut pass, valid)) =
-                            self.compute_passes.remove(&compute_pass_id)
-                        {
+                        if let Pass::Open { mut pass, valid } = pass.take() {
+                            // We want to do steps 1-4 and only if we reached far enough (end returns ok)
```

Comment:
I usually write those like `end` to make clear its a reference to code.

also I think it would be clearer to write: "`pass.end` does step 1-4, and if it returns ok we check the validity of the pass at step 5". with a link to the spec on the first line above.